### PR TITLE
Stability improvement:

### PR DESCRIPTION
**Pull request by jholzer**
Processing in separate threat may cause racing condition and timeouts in browser.
By putting processing in original threat we get a more sequential processing and more stability.

### CODE REVIEW

**Comment by JohnnyCrazy** [comment]
Isn't it blocking the UI Thread then?

Will have a look at it at the weekend
Thanks!

**Comment by jholzer** [comment]
If the server is started on UI thread: yes!
But a server shouldn´t be started on UI-thread anyway...
Since it´s a library i wouldn´t make too much thoughts about this, since it´s in the applications´s responsibility to make the call on the right thread.

Small note: in my application i have a kind of "authentication factory" which delivers the authentication token in a awaitable call. The method takes care about starting the server, sending the request, handle the response and transform it into a result which is then returned to the user once it has completed.
I´m thinking about providing it to your library also. Would be easier for all...

**Comment by jholzer** [comment]
As mentioned above: 
AuthenticationFactory is available now on my branch.
So 1 method call is sufficient to get a fully operational WebApi

**Comment by JohnnyCrazy** [comment]
Nice, I like the idea @jholzer 

Some thoughts:
Since we break the current version anyway, I think we should implement some abstraction into it. Imagine a abstract `AuthFactory`-class, which is the super-class of `ImplicitAuthFactory`, `AuthorizationAuthFactory` and `CredentialsAuthFactory`. Both parse the response and create their tokens. Both have to also implement a method like you already did `SpotifyWebAPI WaitForAuth()` which returns an async task etc... This way we support all 3 methods with abstraction and very easy handling.

What do you think about that?

**Comment by jholzer** [comment]
Hmmm... The problem is, that "AuthenticationFactory" in fact return a finished "SpotifyWebAPI" object while "ClientCredentials" just return a "Token", so does "AutorizationCodeAuth". 
So what would be the common elements for the base-class?
It maybe would make sense for "ClientCredentials"and "AutorizationCodeAuth", since they both handle "Tokens"

Maybe the name "AuthenticationFactory" is a little missleading. 
Should be named e.g. like "WebApiFactory".

**Comment by jholzer** [comment]
Hmmm... The problem is, that "AuthenticationFactory" in fact return a finished "SpotifyWebAPI" object while "ClientCredentials" just return a "Token", so does "AutorizationCodeAuth". 
So what would be the common elements for the base-class?
It maybe would make sense for "ClientCredentials"and "AutorizationCodeAuth", since they both handle "Tokens"

Maybe the name "AuthenticationFactory" is a little missleading. 
Should be named e.g. like "WebApiFactory".

File: .gitignore [hunk 1]
```diff
@@ -1,112 +1,5 @@
-# Build Folders (you can keep bin if you'd like, to store dlls and pdbs)
```
Comment (JohnnyCrazy) [inline .gitignore]: Could you restore this file? I guess VS accidently overwrote it.

**Comment by JohnnyCrazy** [comment]
`WebApiFactory` sounds good.

However, I still think we could add another 2 Factories for the other ones, since these could also provide a finished `SpotifyWebAPI`-object. `Credential`-auth needs to make a HTTP-Call --> Return a `SpotifyWebAPI` once finished. `AutorizationCodeAuth` is similar, but it has one more call to Spotify for the exchange-token.

Gonna have a deeper look next week.

File: .gitignore (hunk 1 above)
Comment (jholzer) [inline .gitignore]: Done!

**Comment by jholzer** [comment]
Yes,of course the other stuff should be wrapped in factories.
Just completed pushes of the first one incl. the requested restore and rename.

File: SpotifyAPI/Local/SpotifyLocalAPI.cs [hunk 2]
```diff
@@ -64,9 +64,21 @@ public SpotifyLocalAPI()
         {
             _rh = new RemoteHandler();
 
+            AttachTimer(50);
```
Comment (JohnnyCrazy) [inline SpotifyAPI/Local/SpotifyLocalAPI.cs]: You could make this shorter with:
```
public SpotifyLocalAPI(int timerIntervall = 50)
{
	_rh = new RemoteHandler();
	AttachTimer(timerIntervall);
}
```

File: SpotifyAPI/Web/Auth/WebApiFactory.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+namespace SpotifyAPI.Web.Auth
+{
+    public class WebApiFactory
+    {
+        private readonly string m_RedirectUrl;
+        private readonly int m_ListeningPort;
+        private readonly string m_ClientId;
+        private readonly TimeSpan m_Timeout;
+        private Scope m_Scope;
```
Comment (JohnnyCrazy) [inline SpotifyAPI/Web/Auth/WebApiFactory.cs]: Could you adapt this names to the code-style?
`private readonly string _redirectUrl`

**Comment by JohnnyCrazy** [comment]
I added some comments mostly relating style-choices.
But it looks very good, gonna merge this after the comments have been cleared. :+1: 

Regarding other Factories:
You were right, the other auth-methods behave very differently. I would suggest we currently stay at one `WebAPIFactory` and may expand it in the future.

File: SpotifyAPI/Web/Auth/WebApiFactory.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using SpotifyAPI.Web.Enums;
+using SpotifyAPI.Web.Models;
+
+namespace SpotifyAPI.Web.Auth
+{
+    public class WebApiFactory
```
Comment (JohnnyCrazy) [inline SpotifyAPI/Web/Auth/WebApiFactory.cs]: Also, this should be `WebAPIFactory` (as well as the filename).

Comment (jholzer) [inline SpotifyAPI/Web/Auth/WebApiFactory.cs]: Just commited the code style changes as requested. (Most were due to my ReSharper-settings... ;-) )

Comment (JohnnyCrazy) [inline SpotifyAPI/Web/Auth/WebApiFactory.cs]: I'm mostly following the default ReSharper settings as well, but the `API` lasted from the beginning. :stuck_out_tongue_winking_eye: 

Gonna merge this tomorrow :+1:

**Comment by JohnnyCrazy** [comment]
Also added a note at the [docs](http://johnnycrazy.github.io/SpotifyAPI-NET/SpotifyWebAPI/auth/#implicitgrantauth)

**Pull request merged**